source/Ship.cpp

```diff
@@ -2877,24 +3051,68 @@ void Ship::Recharge(bool atSpaceport)
 
 bool Ship::CanRefuel(const Ship &other) const
 {
-	return (fuel - JumpFuel(targetSystem) >= other.JumpFuelMissing());
+	// Can't refuel if current ship has zero fuel capacity.
+	if(attributes.Get("fuel capacity") < 1 || GetSystem() != other.GetSystem() || IsEnemyInEscortSystem())
+		return false;
+	if(CanBeCarried())
+	{
+		// Ensure all escorts have minimum one jump of fuel before refueling all
```

Comment:
I don't see why this logic is necessary. First come first serve is perfectly ok imo. If a carried ship asks to refuel first then it should be refueled by the carrier and if then if an escort also needs fuel then too bad, it can ask another ship.

Reply:
To be clear I do not intend to change this at the moment.  It is heavily play tested and the implemented logic feels right.  I encourage you to also play test it to see how the logic feels from a play-ability perspective.  I arrived at this after extensive play testing.

Reply:
> If a carried ship asks to refuel first then it should be refueled by the carrier

This doesn't make sense considering my implementation.  The carrier is not doing any refueling.  The carried ships are doing the refueling.  They act as shuttles between the carrier and other ships.  If carried ships ask for fuel then you have your carried ships sharing fuel with each other and never refueling the fleet.  This is not only inefficient it would also look like strange AI behavior to the player watching carried ships (fighters or drones) refuel each other in a big circle over and over again.

Reply:
This is intended behavior.  I am resolving this feedback.